Lib/test/test_peg_generator/test_pegen.py

```diff
@@ -540,6 +540,33 @@ def test_missing_start(self) -> None:
         with self.assertRaises(GrammarError):
             parser_class = make_parser(grammar)
 
+    def test_invalid_rule_name(self) -> None:
+        grammar = """
+        start: a b
+        _a: 'a'
+        b: 'b'
+        """
+        with self.assertRaises(GrammarError):
```

Comment:
I'd double check here that the GrammarError actually points to the `_a: 'a'` line -- I've had tests like this were the code was broken but the test still passed because the brokenness still caused the same (very general) exception to be raised.